libs/api-provider-environment/src/transaction/useTransactionQueue.tsx

```diff
@@ -0,0 +1,201 @@
+import {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import {
+  TransactionItemStatus,
+  TransactionPayload,
+} from '@webb-tools/webb-ui-components';
+import {
+  Transaction,
+  TransactionState,
+  TransactionStatusValue,
+} from '@webb-tools/abstract-api-provider';
+import { TokenIcon } from '@webb-tools/icons';
+
+function transactionItemStatusFromTxStatus<Key extends TransactionState>(
+  txStatus: TransactionState
+): TransactionItemStatus {
+  switch (txStatus) {
+    case TransactionState.Done:
+    case TransactionState.Ideal:
+      return 'completed';
+    case TransactionState.Failed:
+      return 'warning';
```

Comment:
Should this error if it failed?

Reply:
this is following the design of the `TransactionProgressCard`